src/hex_tarball.erl

```diff
@@ -311,8 +311,20 @@ unpack_tarball(ContentsBinary, Output) ->
     end.
 
 touch(Path) ->
+    case do_touch(Path) of
+        ok -> ok;
+        {error, enoent} -> touch_new(Path)
```

Comment:
previous Elixir code worked because `File.touch` does the same thing under the hood: https://github.com/elixir-lang/elixir/blob/v1.6.4/lib/elixir/lib/file.ex#L493